rskj-core/src/test/java/org/ethereum/net/rlpx/HandshakeHandlerTest.java

```diff
@@ -98,6 +107,13 @@ void shouldDisconnectIfRskCapabilityIsMissing() throws Exception {
         assertFalse(ch.isOpen());
     }
 
+    @Test
+    void shouldConnectWithSnapCapability() throws Exception {
+        simulateHandshakeStartedByPeer(Arrays.asList(new Capability(SNAP, (byte) SNAP_VERSION), new Capability(RSK, EthVersion.UPPER)));
+        // this will only happen when an exception is raised
+        assertTrue(ch.isOpen());
+    }
+
```

Comment:
suggestion:

Here seems a good place to add a test with variations of the capabilities. Like, one with RSK without SNAP and check if the channel will be closed if a handshake with SNAP is attempted. Another without RSK and only with SNAP, where the channel for SNAP should also be closed.

Reply:
Addressed by adding a new test, `shouldDisconnectWithSnapCapabilityIfRskCapabilityIsMissing()`, to ensure that the Handshake doesn't complete if the SNAP capability is present but the RSK capability is missing